### PR TITLE
UserProfile Page, Refactor Profile to React, user can view own posts

### PR DESCRIPTION
**Pull request by lazdinst**
#69, #68, #74 

- Added inline JS styling for profile
- Added a file for map styling; Not being called yet.

### CODE REVIEW

File: client/src/components/main/PostList.jsx [hunk 1]
```diff
@@ -4,14 +4,24 @@ import { connect } from 'react-redux';
...
-      })}
-    </div>
-  );
+const PostList = ({posts, type}) => {
+  if (type === 'TYPE_PROFILE') {
+    return (
+      <div>
+        {posts.map((post, i) => { 
+          return <PostListEntry post={post} id={i} key={i} />; 
+        }).slice(0, 3)}
```
Comment (kevinzhuang10) [inline client/src/components/main/PostList.jsx]: Why is this being sliced?

File: client/src/components/main/UserProfilePage.jsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+          <PostList type={'TYPE_PROFILE'}/>
+          <div className="pull-right">
+            <a href="#">Load more ...</a>
+          </div>
+        </Col>
+      </Row>
+    </div>
+  );
+};
+
```
Comment (Sam0064) [inline client/src/components/main/UserProfilePage.jsx:36]: Eventually we may want to use the React-Bootstrap grid here.

File: server/routes/auth.js [hunk 3]
```diff
@@ -7,40 +7,31 @@ router.route('/')
...
+    Promise.all([models.Post.getAllPosts(), models.Post.getPostsByUserId(req.user.id)])  
+      .then((results) => {
+        preloadedState.posts = results[0];
+        preloadedState.userPosts = results[1];
+      })
+      .then(() => { 
         res.render('index', {preloadedState});
+      })
+      .catch((err) => {
+
```
Comment (kevinzhuang10) [inline server/routes/auth.js]: What's this catch block doing? It's not doing anything. Can we take it out?

File: client/src/components/main/UserProfilePage.jsx (hunk 2 above)
Comment (lazdinst) [reply client/src/components/main/UserProfilePage.jsx:36]: React Bootstrap Grid was generating a "div class="container" element which was messing up the styling of the user page.

File: client/src/components/main/PostList.jsx (hunk 1 above)
Comment (lazdinst) [reply client/src/components/main/PostList.jsx]: This is a naive approach to displaying only the first three posts.

**Review by Sam0064 — DISMISSED** [review]
Approved. Guys lets focus on smaller PR's in the future.

File: server/routes/auth.js (hunk 3 above)
Comment (lazdinst) [reply server/routes/auth.js]: Corrected with commit. https://github.com/mustachioed-meerkats/mustachioed-meerkats/pull/83/commits/7324033b6eaa963a82488d0bbacb61765db36cda

**Review by kevinzhuang10 — DISMISSED** [review]

**Review by kevinzhuang10 — APPROVED** [review]

**Pull request merged**